src/federation-info-plugin/views/ServicesView/ServicesView.jsx

```diff
@@ -1,34 +1,52 @@
-import React from 'react'
+import React, { useMemo, useState } from 'react'
 import { Box } from '@mui/material'
 
 import PanelTitle from '../../components/PanelTitle/PanelTitle'
 import ServiceInfo from '../../components/ServiceInfo/ServiceInfo'
-
+import SearchServiceInput from '../../components/SearchServiceInput/SearchServiceInput'
+import filterServicesInfo from '../../utils/filterServicesInfo'
+import { useDebounce } from 'use-debounce'
 /**
  *
  * @param {Object} props.federationServices result of prepareServicesViewData
  * @returns {JSX.Element}
  */
-const ServicesView = ({ federationServices }) => (
-  <Box
-    sx={{
-      display: 'flex',
-      flexDirection: 'column',
-      minWidth: '200px',
-      flex: 1
-    }}
-  >
-    <PanelTitle total={federationServices.length}>Services</PanelTitle>
-    <Box sx={{ overflow: 'auto', paddingRight: 2 }}>
-      {federationServices.map(({ serviceName, itemsMap }) => (
-        <ServiceInfo
-          key={serviceName}
-          serviceName={serviceName}
-          itemsMap={itemsMap}
-        />
-      ))}
+const ServicesView = ({ federationServices }) => {
+  const [query, setQuery] = useState('')
+  const [defferedQuery] = useDebounce(query, 400)
+
+  const filteredServices = useMemo(
+    () => filterServicesInfo(federationServices, query),
+    [defferedQuery]
```

Comment:
hook missing dependencies, should be `[defferedQuery, federationServices, query]`